migration/migrator/migrations/course/20190722110112_grade_inquiry_per_component.py

```diff
@@ -0,0 +1,21 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    # add gc_id (gradeable compoenent id) foriegn key column to regrade discussion table
+    database.execute("ALTER TABLE regrade_discussion ADD COLUMN gc_id integer")
+    database.execute("ALTER TABLE regrade_discussion ADD CONSTRAINT gradeable_component_id_fk FOREIGN KEY (gc_id) REFERENCES gradeable_component (gc_id)")
+
+    # add eg_is_gi_per_component column to electronic_gradeable table
+    database.execute("ALTER TABLE electronic_gradeable ADD COLUMN eg_is_gi_per_component boolean DEFAULT false NOT NULL")
+    # add a check constraint to make sure that eg_regrade_allowed is true
+    database.execute("ALTER TABLE electronic_gradeable ADD CONSTRAINT regrade_allowed_true_check CHECK (eg_regrade_allowed is true or eg_is_gi_per_component is false)")
+
+def down(config, database, semester, course):
+    # remove gc_id column from regrade_discusion table
+    database.execute("ALTER TABLE regrade_discussion DROP CONSTRAINT gradeable_component_id_fk")
+    database.execute("ALTER TABLE regrade_discussion DROP COLUMN gc_id")
```

Comment:
We don't need to drop the data.  
If we roll back on production we don't want to lose information unless absolutely necessary.

You'll then need to modify the up migration to add column *if not exists* (see other migrations for samples)